### ISSUE TITLE
JDK !7 type handler overhaul

### ISSUE THREAD

**Issue by fh-ms** [eclipse-serializer/serializer#21]
The create method in AbstractBinaryHandlerGenericImmutableCollections12 contains a dangerous hack that should be replace by a proper solution. Or more precisely: a necessary change to how Lazy references are linked to the clustered storage will make this hack work no more. So at the very latest with the clustered MicroStream version, it will have to be fixed.

Some background:

The concept of the loading process and implementation of ALL other BinaryHandlers is the following:
- The loader sends a set of ObjectIds to the PersistenceSource (usually the storage layer).
- It receives a block of binary data back.
- The TypeHandlers know how to make sense of the data (entity headers, primitive values, reference ObjectIds)
- entity instances are created via PersistenceTypeHandler#create without references, just to have an instance to register for each of their ObjectIds.
- The loader uses PersistenceTypeHandler#iterateLoadableReferences to iterate all reference ObjectIds on the BINARY level (the raw block of binary data) and checks which ones are already accounted for and which ones have to be loaded
- This continues iteratively until all raw data objectIds have been resolved and all required raw data is available
- THEN, when it is guaranteed that all instances for occuring objectIds are present, PersistenceTypeHandler#updateState is called to fill in the missing references ("connect" the entities to each other) to the created shallow instances.

There code in the mentioned TypeHandler, however, abuses the current linking workaround Persister instance to execute loading directly from the create method in a type handler. This MIGHT work correctly for now and/or for simple cases (or maybe it even doesn't but an inconsistency caused by it has not been discovered), but it definitely breaks the technical concept, introduces unnecessary complexity and creates an accident waiting to happen should the technical concept change in the slightest way that does not take into account this hack.

Consider:
This hack starts a new loading process WHILE processing an ongoing loading process.
It completely bypasses all the already loaded data, the already resolved objectId, the already created instances.
This surely works in a trivial development test, but if the two (or potentially even more) concurrent and competing(!) loading processes yield different data, any kind of inconsistent data and/or JVM crashes can be the consequence.
In short: NEVER start a new loading process from within an ongoing loading process.

Therefore, this handler should be implemented in the proper way, like the other handlers are:
- #create only creates reference-less instances
- #iterateLoadableReferences iterates the reference's objectIds in the binary data
- #updateState fills in the missing references
